commands/template_store_list.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) OpenFaaS Author(s) 2018. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// DefaultTemplatesStore is the URL where the official store can be found
+	DefaultTemplatesStore = "https://raw.githubusercontent.com/openfaas/store/master/templates.json"
+)
+
+var (
+	templateStoreURL string
+)
+
+func init() {
+	templateStoreListCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "Shows additional language and platform")
+	templateStoreListCmd.PersistentFlags().StringVarP(&templateStoreURL, "url", "u", DefaultTemplatesStore, "Use as alternative store for templates")
+
+	templateStoreCmd.AddCommand(templateStoreListCmd)
+}
+
+// templateStoreListCmd lists templates from default store or custom store if set
+var templateStoreListCmd = &cobra.Command{
+	Use:     `list`,
+	Short:   `List templates from OpenFaaS organizations`,
+	Aliases: []string{"ls"},
+	Long:    `List templates from official store or from custom URL`,
+	Example: `  faas-cli template store list
+  faas-cli template store ls
+  faas-cli template store ls --url=https://raw.githubusercontent.com/openfaas/store/master/templates.json
+  faas-cli template store ls --verbose=true`,
+	RunE: runTemplateStoreList,
+}
+
+func runTemplateStoreList(cmd *cobra.Command, args []string) error {
+	envTemplateRepoStore := os.Getenv(templateStoreURLEnvironment)
+	storeURL := getTemplateStoreURL(templateStoreURL, envTemplateRepoStore, DefaultTemplatesStore)
+
+	templatesInfo, templatesErr := getTemplateInfo(storeURL)
+	if templatesErr != nil {
+		return fmt.Errorf("error while getting templates info: %s", templatesErr)
+	}
+
+	formattedOutput := formatCommandOutput(templatesInfo)
+
+	fmt.Fprintf(cmd.OutOrStdout(), "%s", formattedOutput)
+
+	return nil
+}
+
+func getTemplateInfo(repository string) ([]TemplateInfo, error) {
+	req, reqErr := http.NewRequest(http.MethodGet, repository, nil)
+	if reqErr != nil {
+		return nil, fmt.Errorf("error while trying to create request to take template info: %s", reqErr.Error())
+	}
+
+	client := http.DefaultClient
+	res, clientErr := client.Do(req)
```

Comment:
Do we want to add a timeout to this request?

One way to do this is 

```golang
ctx, cancel := context.WithTimeout(req.Context(), 5*time.Second)
defer cancel()

req = req.WithContext(ctx)
```

Reply:
Yes a timeout makes sense.

Reply:
Ok adding it thanks for showing me how to do it